### DISCUSSION TITLE
Issue: Templated headers don't show up in content Table of Contents

### DISCUSSION THREAD

**Discussion by blockjoe** [McShelby/hugo-theme-relearn#323] [General]
When customizing the `content-footer.html` partial, I'm trying to include a table of contents for any section/chapter pages that currently exist. However, even if I put the anchor and id on my h2/h3 elements, the table of contents remains empty.

For example, this is what is in my `content-footer.html`

```html
{{- if not (.Params.ignore_contents) }}
{{- if .IsSection }}
<h2 id="section-contents">
  Section Contents
  <span class="anchor" title="Copy link to cipboard" data-clipboard-text="{{ .Permalink }}#section-contents">
    <i class="fas fa-link fa-lg" ></i>
  </span>
</h2>
<div>
{{ partial "shortcodes/children.html" (dict
  "context" .
  "description" "true"
  "containerstyle" "div"
  "style" "h3"
)}}
</div>
{{- end }}
{{- end }}
```

But when my page renders, the "Section Contents" entry is not available in the page's table of contents.

![missing-section-contents](https://user-images.githubusercontent.com/69400937/185999351-3c4f6312-b70c-4eaa-9f48-bee523e4dbae.png)

Is it possible to link into the page table of contents creation from other templates in any way?

**Comment by McShelby** [comment]
This is a limitiation of Hugo itself and not possible as far as I know.

The reason is, that your text must be part of the markdown before it is converted to HTML by Hugo.

One way to achieve this, would mean to add a self-written shortcode at the end of each of your markdown files.

**Reply by blockjoe** [reply]
I figured there was a chance that this was something related to how Hugo builds the site, and not how the theme was configured. 

Thanks for clearing up this is something with hugo and not something with the template execution order of the theme!

**Reply by McShelby** [reply]
There may be a trick or two that I am not aware of. So another idea would be to ask for help in the [official Hugo forum](https://discourse.gohugo.io).
